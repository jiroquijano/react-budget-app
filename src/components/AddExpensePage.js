import React from 'react';
import ExpenseForm from './ExpenseForm';
import {connect} from 'react-redux';
import {startAddExpense} from '../actions/expenses';

export class AddExpensePage extends React.Component{
    onSubmit = (expense) =>{
        this.props.addExpenseDispatch(expense);
        this.props.history.push("/");
    }

    render(){
        return (
            <div>
                <h1>Expense page</h1>
                <ExpenseForm onSubmit={this.onSubmit}/>
            </div>
        );
    };
}

// const AddExpensePage = (props) =>(
//     <div>
//         <h1>Expense page</h1>
//         <ExpenseForm onSubmit={(expense)=>{
//             //props.dispatch(addExpense(expense)); 
//             props.onSubmit(expense); //thanks to mapDispatchToProps from react-redux
//             props.history.push("/");
//         }}/>
//     </div>
// );

const mapDispatchToProps = (dispatch)=>({
        addExpenseDispatch: (expense) => dispatch(startAddExpense(expense))
});

const ConnectedAddExpensePage = connect(undefined,mapDispatchToProps)(AddExpensePage);

export default ConnectedAddExpensePage;