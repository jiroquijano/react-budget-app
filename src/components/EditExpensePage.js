import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {startEditExpense,startRemoveExpense} from '../actions/expenses';

export class EditExpensePage extends React.Component{

    onSubmit = (expense) => {
        this.props.editExpenseDispatch(this.props.expense.id, expense);
        this.props.history.push("/");
    }

    onClick = () =>{
        this.props.removeExpenseDispatch(this.props.expense.id);
        this.props.history.push("/");
    }

    render(){ 
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">
                            Edit Expense Page for id: <span>{this.props.expense.id}</span>
                        </h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm 
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button className="button button--secondary" onClick={this.onClick}>Remove Expense</button>
                </div>
            </div>);
    };
};

const mapStateToProps = (state,props) =>({
    expense: state.expenses.find((expense)=>expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch) =>({
    editExpenseDispatch : (id, expense) => dispatch(startEditExpense(id,expense)),
    removeExpenseDispatch : (id) => dispatch(startRemoveExpense(id))
})

const ConnectedEditExpensePage = connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);

export default ConnectedEditExpensePage;