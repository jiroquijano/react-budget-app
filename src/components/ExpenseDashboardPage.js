import React from 'react';

const ExpenseDashboardPage = () =>(
    <div>
    This is from the jiro's dashboard component
    </div>
);

export default ExpenseDashboardPage;