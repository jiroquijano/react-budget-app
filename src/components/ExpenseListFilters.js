import React from 'react';
import {connect} from 'react-redux'
import {setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate} from '../actions/filters';
import {DateRangePicker} from 'react-dates';

export class ExpenseListFilters extends React.Component{
    state = {
        calendarFocused : null
    };

    onDatesChange = ({startDate, endDate}) =>{
        this.props.setStartDateDispatch(startDate);
        this.props.setEndDateDispatch(endDate);
    };

    onFocusChange = (calendarFocused) =>{
        this.setState(()=>({
            calendarFocused
        }));
    };

    dispatchSetTextFilter = (event) => {
        this.props.setTextFilterDispatch(event.target.value);
    }

    sortByFilter = (event) => {
        event.target.value === 'date' ? this.props.sortByDateDispatch() : this.props.sortByAmountDispatch();
    }

    render(){
        return(
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input 
                            className="text-input"
                            type="text"
                            value={this.props.filters.text}
                            onChange={this.dispatchSetTextFilter}
                            placeholder="Search expenses"
                        />
                    </div>

                    <div className="input-group__item">
                        <select
                            className="select"
                            value={this.props.filters.sortBy}
                            onChange={this.sortByFilter}
                        >
                            <option value = "date">Date</option>
                            <option value = "amount">Amount</option>
                        </select>
                    </div>

                    <div className="input-group__item">
                        <DateRangePicker
                            startDate = {this.props.filters.startDate}
                            startDateId = "startdate"
                            endDate = {this.props.filters.endDate}
                            endDateId= "enddate"
                            onDatesChange = {this.onDatesChange}
                            focusedInput = {this.state.calendarFocused}
                            onFocusChange = {this.onFocusChange}
                            showClearDates = {true}
                            numberOfMonths = {1}
                            isOutsideRange = {()=>false}
                        />
                    </div>
                </div> 
            </div>
        );
    };
};

const mapStateToProps = (state) => ({
    filters: state.filters
});

const mapDispatchToProps = (dispatch) =>({
    setTextFilterDispatch : (filter)=> dispatch(setTextFilter(filter)),
    sortByDateDispatch : () => dispatch(sortByDate()),
    sortByAmountDispatch : () => dispatch(sortByAmount()),
    setStartDateDispatch : (startDate) => dispatch(setStartDate(startDate)),
    setEndDateDispatch : (endDate) => dispatch(setEndDate(endDate))
})

const ConnectedExpenseListFilters = connect(mapStateToProps,mapDispatchToProps)(ExpenseListFilters);

export default ConnectedExpenseListFilters;