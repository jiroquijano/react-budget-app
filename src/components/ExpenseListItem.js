import React from 'react';
import {connect} from 'react-redux';
import {removeExpense} from '../actions/expenses';

const ExpenseListItem = ({id, description, amount, createdAt,dispatch}) => (
    <div>
        <h3>{description}</h3>
        <p>Amount: {amount} | Created at: {createdAt}</p>
        <button onClick={()=>{
            dispatch(removeExpense(id));
        }}>Remove</button>
    </div>
);

const ConnectedExpenseListItem = connect()(ExpenseListItem);

export default ConnectedExpenseListItem;