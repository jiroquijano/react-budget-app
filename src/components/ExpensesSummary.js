import React from 'react';
import {connect} from 'react-redux';
import getVisibleExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';
import {Link} from 'react-router-dom';

export class ExpensesSummary extends React.Component{
    render(){
        return (
            <div className="page-header">
                <div className="content-container">
                    <h1 className="page-header__title">Viewing <span>{this.props.expenseCount}</span> {this.props.expenseCount > 1 ? 'expenses' : 'expense'} totalling <span>{numeral(this.props.expensesTotal/100).format('$0,0.00')}</span> </h1>
                    <div className="page-header__actions">
                        <Link className="button" to='/create'>Add Expense</Link>
                    </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) =>{
    const expenses = getVisibleExpenses(state);
    return {
        expenseCount : expenses.length,
        expensesTotal : getExpensesTotal(expenses)
    };
};

const ConnectedExpensesSummary = connect(mapStateToProps)(ExpensesSummary);
export default ConnectedExpensesSummary;