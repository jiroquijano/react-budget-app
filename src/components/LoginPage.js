import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';

export const LogInPage = (props) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Budget App</h1>
            <p>it's time to get your budget under control</p>
            <button className="button" onClick={props.startLogin}> Login with Google </button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin : () => dispatch(startLogin())
});

const ConnectedLoginPage = connect(undefined, mapDispatchToProps)(LogInPage);

export default ConnectedLoginPage;