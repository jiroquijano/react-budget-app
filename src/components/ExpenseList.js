import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import getVisibleExpenses from '../selectors/expenses';

export const ExpenseList = (props) =>(
    <div>
        <h1>Expense List</h1>
        {props.expenses.map(expense=>{
            return (
                <ExpenseListItem
                    key={expense.id} 
                    {...expense}
                />
            );
        })}
    </div>
);

const mapStatetoProps = (state)=>{
    return {
        expenses: getVisibleExpenses(state)
    };
};

export default connect(mapStatetoProps)(ExpenseList);
