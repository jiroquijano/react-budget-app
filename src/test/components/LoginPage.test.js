import React from 'react';
import {shallow} from 'enzyme';
import {LogInPage} from '../../components/LoginPage';

test ("should render LoginPage properly", ()=>{
    const wrapper = shallow(<LogInPage startLogin={()=>{}}/>);
    expect(wrapper).toMatchSnapshot();
});

test ("should call login prop on click", ()=>{
    const loginSpy = jest.fn();
    const wrapper = shallow(<LogInPage startLogin={loginSpy}/>);
    wrapper.find('button').simulate('click');
    expect(loginSpy).toHaveBeenCalled();
});