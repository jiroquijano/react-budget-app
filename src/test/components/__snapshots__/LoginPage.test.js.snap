// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage properly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Budget App
    </h1>
    <p>
      it's time to get your budget under control
    </p>
    <button
      className="button"
      onClick={[Function]}
    >
       Login with Google 
    </button>
  </div>
</div>
`;
